### PR TITLE
[UPDATE] Update deprecated calls from feature layer samples.

### PR DESCRIPTION
**Pull request by D-R-Smith**
### Description

Modularize and remove deprecated method calls from feature layer samples.

Checklist:
- [ ] Set target branch to main (current release) or v.next (next release)
- [ ] Request a reviewer
- [ ] Assign a reviewer
- [ ] Tag reviewer in comment

### CODE REVIEW

**Comment by D-R-Smith** [comment]
Assigning to @jenmerritt but really superseded by other work.

File: feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java [hunk 1]
```diff
@@ -142,9 +142,9 @@ public void start(Stage stage) {
       // the features visible in the viewpoint extent every time the map is redrawn
       featureLayer.addDoneLoadingListener(() -> {
         if (featureLayer.getLoadStatus() == LoadStatus.LOADED) {
-          mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
+          mapView.drawStatusProperty().addListener((property, oldValue, newValue) -> {
             // if the draw status is in progress, show the progress indicator and set the count label text
-            if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.IN_PROGRESS) {
+            if (newValue == DrawStatus.IN_PROGRESS) {
```
Comment (jenmerritt) [inline feature_layers/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/samples/filter_by_definition_expression_or_display_filter/FilterByDefinitionExpressionOrDisplayFilterSample.java:149]: I can't add comments in the right lines, but I think the listener change is fine except we can remove the progress indicator and have a binding there.

So add this above the feature layer listener:
```
// display progress indicator if drawing is in progress
progressIndicator.visibleProperty().bind(mapView.drawStatusProperty().isEqualTo(DrawStatus.IN_PROGRESS));
```

The comment on line 146 can also be changed and should also be moved to inside the if statement:

```
if (newValue == DrawStatus.IN_PROGRESS) {
   // if the draw status is in progress set the count label text
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java [hunk 2]
```diff
@@ -132,8 +132,8 @@ public void start(Stage stage) {
       // set the starting viewpoint for the map view
       mapView.setViewpoint(new Viewpoint(45.5266, -122.6219, 6000));
       // disable the vbox and show a progress indicator when the map view is drawing (e.g. when fetching caches)
-      mapView.addDrawStatusChangedListener(e -> {
-        boolean drawStatusInProgress = e.getDrawStatus() == DrawStatus.IN_PROGRESS;
+      mapView.drawStatusProperty().addListener((property, oldValue, newValue) -> {
+        boolean drawStatusInProgress = newValue == DrawStatus.IN_PROGRESS;
```
Comment (jenmerritt) [inline feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java]: It's annoying that we can't bind to the 'disabledProperty' of the VBox but I think we should go with:

```
      // display the progress indicator when the map view is drawing (e.g. when fetching caches)
      progressIndicator.visibleProperty().bind(mapView.drawStatusProperty().isEqualTo(DrawStatus.IN_PROGRESS));

      // disable the vbox when the map view is drawing (e.g. when fetching caches)
      mapView.drawStatusProperty().addListener((property, oldValue, newValue) -> 
        controlsVBox.setDisable(newValue == DrawStatus.IN_PROGRESS));
```

**Review by jenmerritt — COMMENTED** [review]
Some suggestions on the use of the properties.

These 3 samples also need modularizing.

File: feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java [hunk 3]
```diff
@@ -84,8 +84,8 @@ public void start(Stage stage) {
       mapView.getSelectionProperties().setColor(0xFFFFFF00);
 
       // hide the progress indicator when the layer is done drawing
-      mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
-        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+      mapView.drawStatusProperty().addListener((property, oldValue, newValue) -> {
+        if (newValue == DrawStatus.COMPLETED) {
```
Comment (jenmerritt) [inline feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java]: We can make this:

```
// show the progress indicator when the layer is drawing
progressIndicator.visibleProperty().bind(mapView.drawStatusProperty().isEqualTo(DrawStatus.IN_PROGRESS));
```

File: feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java (hunk 2 above)
Comment (Luiy0) [reply feature_layers/toggle-between-feature-request-modes/src/main/java/com/esri/samples/toggle_between_feature_request_modes/ToggleBetweenFeatureRequestModesSample.java]: @jenmerritt When the sample loads for the first time, the draw status changes causing the VBox to quickly alternate between enabled/disabled. We could potentially leave it as is or keep the VBox as always enabled if you think the blinking is distracting.

**Comment by jenmerritt** [comment]
Closing this PR for now, will leave the branch for further exploration, as per @Luiy0 comments

**Pull request closed without merging**